src/pytorch_lightning/loggers/wandb.py

```diff
@@ -357,13 +357,14 @@ def experiment(self) -> Run:
                 self._experiment = wandb.init(**self._wandb_init)
 
                 # define default x-axis
-                if getattr(self._experiment, "define_metric", None):
+                if isinstance(self._experiment, Run) and getattr(self._experiment, "define_metric", None):
                     self._experiment.define_metric("trainer/global_step")
                     self._experiment.define_metric("*", step_metric="trainer/global_step", step_sync=True)
 
+        assert isinstance(self._experiment, Run)
```

Comment:
Ah yes, it fails because of the mock object, I'll look into it.

Reply:
I don't think that was it, the tests are still failing

Reply:
I've just fixed all the failing tests now except the failing CLI test which is a bit obscure. 

The problems with the failing tests was that the return type of the `experiment` property was expected to be `Run` and hence why I asserted it, but the logger tests used a `Mock` object instead of a `Run`. 

For the remaining CLI test `test_wandb_logger_init_args`, I don't think the culprit is this added `assert` statement, but not sure what is..

Reply:
@gautierdag I don't remember exactly, but there is a way to make a mock pass the isinstance check. 
I think something like `Mock(spec=Run)` will make it pass `assert isinstance(mock, Run)`.

Reply:
@awaelchli Yep that's one way! However because `wandb` I didn't want the tests to rely on `wandb` being installed (to then import `Run` from), instead I decided to patch the `Run` type as `mock.Mock` where necessary.